user/build-matrix.md

```diff
@@ -0,0 +1,294 @@
+---
+title: Build Matrix
+layout: en
+---
+
+The build matrix is a way to run builds on multiple versions of language runtimes, on multiple operating systems, and with multiple versions of environment variables.
+
+<div id="toc"></div>
+
+When you combine the three main configuration options of *Runtime*, *Environment* and *Exclusions/Inclusions* you have a matrix of all possible combinations.
+
+Each programming language environment on Travis CI has a specific list of keys which automatically populate the build matrix. For Ruby, those are `rvm`, `gemfile`, `env`, and `jdk`.
```

Comment:
Are we enumerating the language-specific expansion keys? If so, `env` should not appear here.

At the same time, the previous paragraph mentions "*Runtime*", so having `rvm` here is also confusing.

Maybe, we can rewrite this paragraph with something like:

> The set of keys which are expanded to create the build matrix varies from one programming language to another. For Ruby, they are `env`, `gemfile`, `jdk`, and `rvm`. These are specifically mentioned in the reference documentation for each language as "matrix keys".

Reply:
:100: